src/main/java/org/springframework/cloud/scheduler/spi/cloudfoundry/CloudFoundryAppScheduler.java

```diff
@@ -48,65 +63,92 @@
  */
 public class CloudFoundryAppScheduler implements Scheduler {
 
-	private final SchedulerClient client;
+	private final static int PCF_PAGE_START_NUM = 1; //First PageNum for PCFScheduler starts at 1.
 
+	protected final static Log logger = LogFactory.getLog(CloudFoundryAppScheduler.class);
+	private final SchedulerClient client;
 	private final CloudFoundryOperations operations;
-
 	private final CloudFoundryConnectionProperties properties;
-
 	private final CloudFoundry2630AndLaterTaskLauncher taskLauncher;
-
-	protected static final Log logger = LogFactory.getLog(CloudFoundryAppScheduler.class);
+	private final CloudFoundrySchedulerProperties schedulerProperties;
 
 	public CloudFoundryAppScheduler(SchedulerClient client, CloudFoundryOperations operations,
-			CloudFoundryConnectionProperties properties, CloudFoundry2630AndLaterTaskLauncher taskLauncher) {
+			CloudFoundryConnectionProperties properties, CloudFoundry2630AndLaterTaskLauncher taskLauncher,
+			CloudFoundrySchedulerProperties schedulerProperties) {
 		Assert.notNull(client, "client must not be null");
 		Assert.notNull(operations, "operations must not be null");
 		Assert.notNull(properties, "properties must not be null");
 		Assert.notNull(taskLauncher, "taskLauncher must not be null");
+		Assert.notNull(schedulerProperties, "schedulerProperties must not be null");
 
 		this.client = client;
 		this.operations = operations;
 		this.properties = properties;
 		this.taskLauncher = taskLauncher;
+		this.schedulerProperties = schedulerProperties;
 	}
 
 	@Override
 	public void schedule(ScheduleRequest scheduleRequest) {
 		String appName = scheduleRequest.getDefinition().getName();
 		String jobName = scheduleRequest.getScheduleName();
```

Comment:
I see that the scheduler API uses `scheduleJob` and hence the `jobName` here is used. Given, we enforce the clear distinction between task and job (being the subset of task) and having the reference to `job` in a different context for scheduler brings some confusion. I am sure this must have been discussed before but, is there any reason why we use `scheduleJob` instead of `scheduleTask`?

Reply:
Yeah I debated that myself.   Updated to use `task` instead of `job`